ipaserver/plugins/cert.py

```diff
@@ -791,8 +793,10 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
                             "match requested principal") % gn.name)
             elif isinstance(gn, cryptography.x509.general_name.RFC822Name):
                 if principal_type == USER:
-                    if principal_obj and gn.value not in principal_obj.get(
-                            'mail', []):
+                    gn_value = (gn.value if isinstance(gn.value, list)
+                                         else [gn.value])
```

Comment:
I think `gn.value` will always be a string, so you can just put:

```python
if not _emails_are_valid([gn.value], principal_obj.get('mail', [])):
```

If I'm wrong about that, could you please explain?